spyder/widgets/panels/classfunctiondropdown.py

```diff
@@ -0,0 +1,406 @@
+
+    @property
+    def range(self):
+        return self.fold_scope.get_range()
+
+    @property
+    def start_line(self):
+        return self.range[0]
+
+    @property
+    def end_line(self):
+        return self.range[1]
+
+    @property
+    def oed(self):
+        return self._oed
+
+    @property
+    def name(self):
+        return self.oed.def_name
+
+    @property
+    def line(self):
+        return self.start_line
+
+    @property
+    def def_type(self):
+        if self.oed.def_type == OED.FUNCTION:
+            return OED.FUNCTION_TOKEN
+        elif self.oed.def_type == OED.CLASS:
+            return OED.CLASS_TOKEN
+        else:
+            return "Unknown"
+
+
+class ClassFunctionDropdown(Panel):
+    """
+    Class and Function/Method Dropdowns Widget.
+
+    Parameters
+    ----------
+    editor :
+        The editor to act on.
+    """
+
+    def __init__(self, editor):
+        Panel.__init__(self, editor)
+        self._editor = editor
+        self._editor.sig_cursor_position_changed.connect(
+            self._handle_cursor_position_change_event
+        )
+
+        # The layout
+        hbox = QHBoxLayout()
+        self.class_cb = QComboBox()
+        self.method_cb = QComboBox()
+        hbox.addWidget(self.class_cb)
+        hbox.addWidget(self.method_cb)
+        hbox.setSpacing(0)
+        hbox.setContentsMargins(0, 0, 0, 0)
+
+        self.setLayout(hbox)
+
+        # Internal data
+        self.folds = None
+        self.parents = None
+        self.classes = None
+        self.funcs = None
+
+        # Initial data for the dropdowns.
+        self.class_cb.addItem("<None>", 0)
+        self.method_cb.addItem("<None>", 0)
+
+        # Attach some events.
+        self.class_cb.activated.connect(self.combobox_activated)
+        self.method_cb.activated.connect(self.combobox_activated)
+
+    def sizeHint(self):
+        """Override Qt method."""
+        return QSize(0, self._getVerticalSize())
+
+    def _getVerticalSize(self):
+        """Get the default height of a QComboBox."""
+        return self.class_cb.height()
+
+    def _update_data(self):
+        """Update the internal data values."""
+        _old = self.folds
+        self.folds = _get_fold_levels(self.editor)
+
+        # only update our dropdown lists if the folds have changed.
+        if self.folds != _old:
+            self.classes, self.funcs = _split_classes_and_methods(self.folds)
+            self.populate_dropdowns()
+
+    def populate_dropdowns(self):
+        self.class_cb.clear()
+        self.method_cb.clear()
+
+        populate(self.class_cb, self.classes)
+        populate(self.method_cb, self.funcs)
+
+    def combobox_activated(self):
+        """Move the cursor to the selected definition."""
+        sender = self.sender()
+        data = sender.itemData(sender.currentIndex())
+
+        # TODO: don't trigger a _handle_cursor_position_change_event
+        self.editor.go_to_line(data.line + 1)
```

Comment:
This line is throwing an error when the `<none>` item is selected, I think and `if` should be added:

```
if data:
    self.editor.go_to_line(data.line + 1)
```